delivery_state/models/stock_picking.py

```diff
@@ -40,7 +41,35 @@ class StockPicking(models.Model):
     )
 
     def tracking_state_update(self):
-        for picking in self.filtered(lambda p: p.delivery_type):
+        """Call to the service provider API wich should have the method
+        definedin the manner. Can be triggered manually or by cron:
+        <my_provider>_tracking_state_update"""
+        for picking in self.filtered("delivery_type"):
             method = '%s_tracking_state_update' % picking.delivery_type
             if hasattr(picking.carrier_id, method):
-                getattr(picking.carrier_id, method)(picking)
+                return getattr(picking.carrier_id, method)(picking)
```

Comment:
if you use return you interrupt the loop

Reply:
you're absolutly right :+1: